### ISSUE TITLE
[469b] UMake can't recognize the main game configuration file

### ISSUE THREAD

**Issue by HalfWayLambda** [OldUnreal/UnrealTournamentPatches#483]
UMake doesn't work with the 469b version of UT. I've made sure that UMake.exe is inside the main root directory of UT (and not just inside the folder System or whatever), and it always gives the error of `Unable to find the main game configuration file`. I can't figure out why it can't find it, since there are no missing files. From what I know the "main configuration file" of UT is `UnrealTournament.ini`, which obviously does exist inside my System folder.
And yes, I tested this with the version 436 and UMake recognized it just fine.

**Comment by SeriousBarbie** [comment]
You can use [Process Monitor](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon) to see what UMake.exe tries to open.

**Comment by HalfWayLambda** [comment]
Ok I ran Process Monitor on it. Apparently it does find all of the files inside the System folder (`UnrealTournament.ini` and all that), but for some reason it decides to just not read them. There is an operation of `ReadFile` on each one, with an offset of 0, meaning that it starts to read the contents of the file, but then it stops, even though there should be plently of more `ReadFile` operations after that with an increasing offset (usually it goes 0, 128, 256, 384, 512 etc). However, this doesn't happen with `DefUser.ini`, which is the only file that gets read correctly. I really don't know what's causing this. I'll try to look a bit deeper though.
I tested this with the 436 version of UT and guess what: UMake reads all of the files correctly, without just stopping at offset 0 (we're talking about the exact same UMake.exe here).

**Comment by stijn-volckaert** [comment]
Could this be a permission problem? Did you install 469 in Program Files perhaps?

**Comment by HalfWayLambda** [comment]
No, I have it installed in E:/Games/UT99_Patched, and the 436 version in E:/Games/UT99. It works with 436 and somehow it does not with 469b..

**Comment by SeriousBuggie** [comment]
What about encoding of UT.ini for v436 and v469b? Maybe v469b use UTF-8, while UMake expect UTF-16LE?
I try copy my UMake executable into v469b directory and run, but it run fine and still show older folder fol last project. So I suspect UMake use some registry settings and you can not reproduce issue condition just copy UMake.exe into new UT folder.

**Comment by HalfWayLambda** [comment]
Oh damn, you're right. v436 uses encoding UTF-8, whereas v649b uses UCS-2 LE BOM. I guess there isn't a way to make UMake recognize UCS-2? I have tried copy-pasting the same exe onto the patched folder, but it still gives out the error. Also I want to be clear: the problem is not that UMake doesn't run, but rather it doesn't compile anything because of the `Unable to find the main game configuration file` error. If I open the program normally (without dragging and dropping a folder onto it), then the `Compile` button is just greyed out.

**Comment by stijn-volckaert** [comment]
This is not fully accurate. Both UT v436 and v469 use your system's ANSI code page to encode the ini file if the ini file does not contain characters whose literal wide-character value is above 255. In other words, unless you're trying to save an ini file with special Unicode characters in it, the file will just be saved in plain ANSI. Furthermore, if all characters in your ini file are in the 1-127 range, the plain ANSI and the UTF-8 encoding will be exactly the same.

If your ini _DOES_ contain Unicode characters (i.e., characters with a literal wchar_t value above 255), the ini will be saved in UTF-16LE/UCS-2 LE BOM encoding. This too is true for UT v436 and v469.

The only real difference regarding encoding between the v436 and v469 Windows binaries is that UT v469 can handle UTF-8-encoded files, while v436 cannot.

**Comment by stijn-volckaert** [comment]
I just tested UMake v1.2.1 myself and it worked fine for me. I tried running it from my desktop and running it from my 469 system folder. I also tried encoding my ini file as ANSI or UTF-16LE. All tests worked fine.

**Comment by HalfWayLambda** [comment]
I see. So there must be one or more special unicode characters in my own UnrealTournament.ini? I tried deleting it, so that the game would create a new one, and that got encoded with UCS-2 LE BOM again. And interestingly, i tried converting it to UTF-8 with notepad++, and UMake worked fine this time! But after i re-launch the game it changes to UCS-2 again and then UMake throws the same error. Does UMake even support this type of encoding? I've downloaded it from many different sources, just in case my version was somehow corrupt, but nothing..
